app/src/main/jni/audiolat.cpp

```diff
@@ -42,36 +43,38 @@ aaudio_data_callback_result_t dataCallback(AAudioStream *stream, void *userData,
   if (stream == cb_data->record_stream) {
     LOGD("record num_frames: %d time_sec: %.2f", num_frames, time_sec);
     // recording
-    if (time_sec - last_ts > 2) {
+    if (time_sec - last_ts > 2 || rec_buffer_index > 0) {
       playing = true;
-      LOGD("record write: input num_frames: %d", num_frames - cb_data->begin_signal_size);
+      LOGD("record write begin signal: input num_frames: %d", num_frames - cb_data->begin_signal_size);
       fwrite((int16_t *)audioData, sizeof(int16_t),
-             (size_t)num_frames - cb_data->begin_signal_size, cb_data->output_file_descriptor);
-      LOGD("record write: begin num_frames: %i", cb_data->begin_signal_size);
-      fwrite(cb_data->begin_signal, (size_t)cb_data->begin_signal_size,
+             (size_t)num_frames - cb_data->begin_signal_size, cb_data->output_file_descriptor);\
```

Comment:
remove backslash at the end of line

Reply:
you're not solving the problem, right? If the number of frames in the record buffer is smaller than the size of the being signal, this would still try to write a negative number of bytes.

Reply:
If the number of frames in the callback buffer is _larger_ than the remaining data to be written write the remaining data otherwise write the number of frames in the buffer. 
How is this not solving the problem?